### PR TITLE
Admin login

### PR DESCRIPTION
**Pull request by ali-7**
## Description

create login front-end and back-end code
create logout function and route for admin

### Type of change

New features #5 
Description : create login.js that contain fetch function that make a request (post) to back-end when the admin enter username and password

New features #6 
Description : 
* create login.js in the controller that contain function that deal with the request from the front-end and get the admin password and compere it with the password that stored in DB. and add cookie to the admin when the password checked.

* create logout.js : that clear the cookie when the admin sign out 
* create isLoggedIn.js : function that check if the current session has a cookie (admin logged in ) or not, and if not it will redirect the admin to sign in again.

New features #15
Description : create checkAdmin query, that return the admin username and password from db.

### CODE REVIEW

**Review by lubnaabd — APPROVED** [review]

File: src/controllers/isLoggedIn.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const { parse } = require('cookie');
```
Comment (RamyAlshurafa) [inline src/controllers/isLoggedIn.js]: use `snake_case` for files naming

File: src/controllers/index.js [hunk 2]
```diff
@@ -4,22 +4,30 @@ const login = require('./login');
...
-router.get('/admin/users', user.getViewUser);
-router.get('/admin/users/borrower', user.getBorrowedUser);
-router.get('/admin/books/library', books.getLibraryBooks);
-router.get('/admin/books/store', books.getStoreBooks);
-router.get('/admin/books/borrowed', books.getBorrowedBooks);
-router.get('/admin/books/add', books.getAddBookTab);
-router.get('/admin/borrow', borrow.get);
+router.post('/admin/login', login.post);
+
+// admin control panel routes
```
Comment (RamyAlshurafa) [inline src/controllers/index.js:22]: you can add `isLoggedIn` as a middleware here (before admin's routes) so no need to add it at every rout
```js
router.use('/admin', isLoggedIn)
```

File: public/js/login.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const username = document.getElementById('username');
+const password = document.getElementById('password');
+const loginButton = document.getElementById('loginButton');
+
+loginButton.addEventListener('click', (e) => {
```
Comment (RamyAlshurafa) [inline public/js/login.js:5]: for better user experience add the listener for the form  on `submit` to accept both `click` and `Enter`

File: src/controllers/test/test.js [hunk 4]
```diff
@@ -75,3 +108,34 @@ test('test for website landing page route ', (t) => {
       t.end();
     });
 });
+
+// // tests for login page
```
Comment (RamyAlshurafa) [inline src/controllers/test/test.js]: `// //` => `//`

File: src/database/queries/checkAdmin.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const dbConnection = require('../db_connection');
+
+const getAdmin = (username, cb) => {
+  const sql = {
+    text: 'SELECT * FROM admin WHERE user_name = $1 ',
+    values: [username],
+  };
+  dbConnection.query(sql, (err, result) => {
+    if (err) return cb(err);
+    return cb(null, result.rows);
```
Comment (RamyAlshurafa) [inline src/database/queries/checkAdmin.js:10]: your admin's `user_name` is `UNIQUE` so you are sure that this query will return a single row
so you can send `result.rows[0]`
so you can use it directly in `test` and `controller`

File: src/controllers/logout.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+exports.get = (request, response) => {
+  response.cookie('data', '0', { maxAge: 0, httpOnly: true });
```
Comment (RamyAlshurafa) [inline src/controllers/logout.js]: you can use this built-in function to clear cookies `res.clearCookie("key");`

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
- Use `snake_case` for files naming
- target the form with `submit` listener
- use authentication middleware once

File: src/database/queries/checkAdmin.js (hunk 5 above)
Comment (ali-7) [reply src/database/queries/checkAdmin.js:10]: @RamyAlshurafa it's like the same thing right ??

**Review by RamyAlshurafa — APPROVED** [review]
Great Job :+1:

**Pull request merged**